### PR TITLE
[dvdnav] Update to current next version

### PR DESCRIPTION
**Pull request by enen92**
## Description
As you might know, I've been trying to get rid of our patches on libdvdnav so that at some point in the future we can simply rely on upstream. It's a tedious job but we manage to get some of them upstream:
https://code.videolan.org/videolan/libdvdnav/-/commit/e8d9d3ae7a0d3dda68f1e0954c2294a7b8d475b6

This allow us to remove the usage of private structs in the lib. You can see the difference between the "future" state and the current one by comparing the commits:
- Next: https://github.com/enen92/libdvdnav/commits/nexus
- Current: https://github.com/xbmc/libdvdnav/commits/master

According to my math we dropped already 3 patches. Might sound residual but they affect big parts of our library handling (especially the `get_vm` patch).
This PR syncs the headers, drops private stuff and updates the dvdnav inputstream to use only public functions from the library.

### CODE REVIEW

**Comment by enen92** [comment]
jenkins build this please

File: tools/depends/target/libdvdnav/LIBDVDNAV-VERSION [hunk 1]
```diff
@@ -1,5 +1,5 @@
 LIBNAME=libdvdnav
 BASE_URL=https://github.com/xbmc/libdvdnav
-VERSION=6.0.0-Leia-Alpha-3
+VERSION=6.1.1-Next-Nexus-Alpha2
 ARCHIVE=$(LIBNAME)-$(VERSION).tar.gz
-SHA512=11c93eaacd156f8fd7dec7c43d366438b201f31ad55b2870463a9e286912b6ada08882319a021fb7992190f87b909a49f2b83e0321cc17aedc29f7fe5898fa72
+SHA512=079edcd7732b4edbf90806d9acc0a03efef386612df48fa711fd944320034630ad89dbdb7d72bdea0560b7dac71229057c02cb8bf0c9d70e1b86680bfd08b84d
```
Comment (howie-f) [inline tools/depends/target/libdvdnav/LIBDVDNAV-VERSION:5]: there seems to be a hash mismatch, calculates
`01c415a561ea8bd9d568ad2668fbf8bc54304557e413776e1b841145cad3f7b9991a6772af7bac408b2cb87c6236c378e79c1c023a58b280a6c92fee151f0fb1` for me

Comment (enen92) [reply tools/depends/target/libdvdnav/LIBDVDNAV-VERSION]: yeah, was hoping the hash would be the same after pushing to the new repo. Fixed

**Comment by enen92** [comment]
yay, finally green. Thanks much paxxi and fuzzard for the help with the windows/cmake stuff.

**Comment by thexai** [comment]
Runtime tested Windows x64 on top of current master branch.
Also tested with https://github.com/xbmc/xbmc/pull/21500

Blu-Ray's menus works fine:

screenshot00004

However, found a issue, I don't know if it's directly related:

When Blu-Ray playback resumes, the message asking whether to resume or start from the beginning does not appear, nor does the menu for selecting the track to play / menu appear.

Plays directly resuming from the last point where it left off and when this happens the "menu" icon does not appear in the OSD during playback:

screenshot00006


When start playing from the beginning (the first time) the menu icon does appear:

menu

**Comment by enen92** [comment]
@thexai this PR doesn't touch blurays only dvds. The menu issue in blurays is something that was broken in previous versions. If you resume or play a track (without going via the menu) the navigation mode is not enabled so despite the menu icon was there it didn't work.
For those cases I plan to enable the simplified menu as an alternative but it's not easy to add since it's another GUI dependency. I am still trying to picture how to best handle that

**Comment by enen92** [comment]
I've tested this in linux, windows and android. Menus and playback are working fine.
Don't think there's a need to bother others to test so I'll just go ahead and merge this.
Thanks to all who have provided help updating this package

**Pull request merged**